ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake

```diff
@@ -33,10 +33,21 @@ function(ament_target_dependencies target)
     message(FATAL_ERROR "ament_target_dependencies() the first argument must be a valid target name")
   endif()
   if(${ARGC} GREATER 0)
+    set(options PUBLIC)
+    set(oneValueArgs)
+    set(multiValueArgs)
+    cmake_parse_arguments(AMENT_TARGET_DEPENDENCIES
+      "${options}"
+      "${oneValueArgs}"
+      "${multiValueArgs}"
+      ${ARGN})
     set(definitions "")
     set(include_dirs "")
     set(libraries "")
     set(link_flags "")
+    if(AMENT_TARGET_DEPENDENCIES_PUBLIC)
+      list(REMOVE_ITEM ARGN ${options})
+    endif()
```

Comment:
You shouldn't manipulate `ARGN` - you can used `AMENT_TARGET_DEPENDENCIES_UNPARSED_ARGUMENTS` instead.